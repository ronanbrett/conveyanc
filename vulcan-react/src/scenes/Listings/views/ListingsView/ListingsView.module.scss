@import "src/styles/vars";

.ListingsView {
  height: calc(100vh - 70px);

  overflow: auto;
  scroll-snap-type: y mandatory;
}
