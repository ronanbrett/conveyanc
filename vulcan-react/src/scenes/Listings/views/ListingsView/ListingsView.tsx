import { PropertyDto } from "@core/api/graphql";
import { ListingStore } from "@scenes/Listings/Listings.store";
import React, { useEffect, useState } from "react";
import { ListingCard } from "../../components";
import styles from "./ListingsView.module.scss";

interface ListingsViewProps {
  children?: any;
}

const ListingsView = (props: ListingsViewProps) => {
  const [listingState, setListingState] = useState(ListingStore.initialState);

  useEffect(() => {
    const sub = ListingStore.subscribe(setListingState);
    ListingStore.init();

    return () => {
      sub.unsubscribe();
    };
  }, []);

  return (
    <div className={`${styles.ListingsView} page`}>
      {listingState.data.map((listing: PropertyDto) => (
        <ListingCard key={listing._id} />
      ))}
    </div>
  );
};

export default ListingsView;
