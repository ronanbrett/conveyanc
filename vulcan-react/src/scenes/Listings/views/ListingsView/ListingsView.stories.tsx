import React from "react";

import ListingsView from "./ListingsView";

import { action } from "@storybook/addon-actions";

export default {
  title: "Components/ListingsView",
  decorators: [],
  component: ListingsView,
};

export const Default = () => <ListingsView></ListingsView>;
