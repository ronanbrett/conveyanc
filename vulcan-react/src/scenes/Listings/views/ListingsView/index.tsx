import ListingsView from './ListingsView';

export default ListingsView;