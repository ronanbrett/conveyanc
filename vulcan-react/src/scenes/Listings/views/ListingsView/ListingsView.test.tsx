import React from "react";
import { render } from "@testing-library/react";
import ListingsView from "./ListingsView";
import { MockRouter, createTopNavPortalMock } from "testing";

createTopNavPortalMock();
const TestViewComponent = () => <MockRouter component={ListingsView} />;

describe("ListingsView", () => {
  it("renders without error", () => {
    const { asFragment } = render(<TestViewComponent />);
    expect(asFragment()).toMatchSnapshot();
  });
});
