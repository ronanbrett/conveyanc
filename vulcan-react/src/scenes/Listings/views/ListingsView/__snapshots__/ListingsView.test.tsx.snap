// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingsView renders without error 1`] = `
<DocumentFragment>
  <div
    class="ListingsView page"
  />
</DocumentFragment>
`;
